### PR TITLE
feat(query): Add `external_block_rows` metrics

### PR DESCRIPTION
**Pull request by b41sh**
I hereby agree to the terms of the CLA available at: https://docs.databend.com/dev/policies/cla/

## Summary

1. Add a new metrics `external_block_rows` to record the number of rows per block when calling the udf servers.
2. Fix the metrics `external_request_duration` and `external_connect_duration` to record data in milliseconds, and fix the metric data loss caused by the wrong record unit.
3. Create `endpoints` in the `init_endpoints` function.
4. Create `semaphore` in the `init_semaphore` function to make max threads permit in query level instead of transform level.

- fixes: #[Link the issue here]

## Tests

- [ ] Unit Test
- [ ] Logic Test
- [ ] Benchmark Test
- [x] No Test - _Explain why_

## Type of change

- [ ] Bug Fix (non-breaking change which fixes an issue)
- [x] New Feature (non-breaking change which adds functionality)
- [ ] Breaking Change (fix or feature that could cause existing functionality not to work as expected)
- [ ] Documentation Update
- [ ] Refactoring
- [ ] Performance Improvement
- [ ] Other (please describe):

- - -
This change is [Reviewable](https://reviewable.io/reviews/databendlabs/databend/17116)

### CODE REVIEW

File: src/query/service/src/pipelines/processors/transforms/transform_udf_server.rs [hunk 1]
```diff
@@ -79,6 +78,20 @@ impl TransformUdfServer {
...
+        ctx: Arc<QueryContext>,
+        funcs: Vec<UdfFunctionDesc>,
+        endpoints: BTreeMap<String, Arc<Endpoint>>,
+    ) -> Result<Self> {
+        let settings = ctx.get_settings();
+        let connect_timeout = settings.get_external_server_connect_timeout_secs()?;
+        let request_batch_rows = settings.get_external_server_request_batch_rows()? as usize;
+        let request_max_threads = settings.get_external_server_request_max_threads()? as usize;
+        let retry_times = settings.get_external_server_request_retry_times()? as usize;
+        let semaphore = Arc::new(Semaphore::new(request_max_threads));
```
Comment (sundy-li) [inline src/query/service/src/pipelines/processors/transforms/transform_udf_server.rs:94]: seems this is not query level but transform level.

Comment (b41sh) [reply src/query/service/src/pipelines/processors/transforms/transform_udf_server.rs]: fixed

**Review by sundy-li — APPROVED** [review]

File: src/common/base/src/runtime/metrics/histogram.rs [hunk 2]
```diff
@@ -39,6 +39,11 @@ pub static BUCKET_MILLISECONDS: [f64; 15] = [
     300000.0, 600000.0, 1800000.0,
 ];
 
+pub static BUCKET_ROWS: [f64; 12] = [
+    1.0, 10.0, 100.0, 1000.0, 10000.0, 50000.0, 100000.0, 500000.0, 1000000.0, 5000000.0,
```
Comment (flaneur2020) [inline src/common/base/src/runtime/metrics/histogram.rs:43]: add a 5000 between 1000 and 10000?

**Review by flaneur2020 — APPROVED** [review]
LGTM :+1:

**Comment by sundy-li** [comment]
image

Perf test is to show performance with the latest release. 

It is not related to this pr; it is also seen in the latest changes.

**Pull request merged**